src/value.rs

```diff
@@ -66,18 +67,21 @@ impl NoteValue {
 }
 
 impl Sub for NoteValue {
-    type Output = Result<ValueSum, OverflowError>;
+    type Output = Option<ValueSum>;
 
+    #[allow(clippy::suspicious_arithmetic_impl)]
     fn sub(self, rhs: Self) -> Self::Output {
-        let a: i64 = self.0.try_into().map_err(|_| OverflowError)?;
-        let b: i64 = rhs.0.try_into().map_err(|_| OverflowError)?;
-        Ok(ValueSum(a - b))
+        let a = self.0 as i128;
+        let b = rhs.0 as i128;
+        a.checked_sub(b)
+            .filter(|v| v > &(-(std::u64::MAX as i128)) && v < &(std::u64::MAX as i128))
```

Comment:
```suggestion
            .filter(|v| v >= &(-(u64::MAX as i128)) && v =< &(u64::MAX as i128))
```
- The protocol spec defines the valid range as {−2<sup>64</sup>+1..2<sup>64</sup>−1} where {a..b} is inclusive at both ends.
- `u64::MAX` is defined as 2<sup>64</sup>−1.
- Also, `std::u64::MAX` is deprecated.

This might be easier to read as:
```suggestion
            .filter(|v| VALUE_SUM_RANGE.contains(v))
```

with this at the top:
```rust
use std::ops::RangeInclusive;

/// The valid range of the scalar multiplication used in ValueCommit^Orchard.
///
/// Defined in a note in [Zcash Protocol Spec § 4.17.4: Action Statement (Orchard)][actionstatement].
///
/// [actionstatement]: https://zips.z.cash/protocol/nu5.pdf#actionstatement
const VALUE_SUM_RANGE: RangeInclusive<i128> = -(u64::MAX as i128)..=u64::MAX as i128;
```